modules/auxiliary/gather/query_ldap.rb

```diff
@@ -0,0 +1,279 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::LDAP
+  require 'json'
+  require 'yaml'
+
+  def initialize(info = {})
+    begin
+      @default_settings_file_path = ::File.join(::Msf::Config.data_directory, 'auxiliary', 'gather', 'query_ldap', 'default.yaml')
+      @default_settings = YAML.safe_load_file(@default_settings_file_path)
+    rescue StandardError => e
+      print_error("Couldn't parse #{@default_settings_file_path}, error was: #{e}")
+      return
+    end
+
+    unless @default_settings['queries']&.class == Array && !@default_settings['queries'].empty?
+      print_error("No queries supplied in #{@default_settings_file_path}!")
+      return
+    end
+
+    actions = []
+    for entry in @default_settings['queries']
+      if entry['action'].nil? || entry['description'].nil?
+        print_warning("Invalid entry detected, check the format of the file at #{@default_settings_file_path}!")
```

Comment:
I'm not sure if this is enough information for a user to work out what the problem is in isolation, the error message may need to be a bit more descriptive.

Reply:
Good point let me update this to make it easier to use for debugging 👍

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/16598/commits/b44613113cd8e95ad0798d58138ad86e98de2e07 with a better description and some logic to print out the bad action name if available which should help with debugging.